src/rust/engine/graph/src/entry.rs

```diff
@@ -230,6 +230,14 @@ impl<N: Node> Entry<N> {
     &self.node
   }
 
+  pub(crate) fn cacheable_with_output(&self, output: Option<&N::Item>) -> bool {
+    (if let Some(item) = output {
+      self.node.cacheable_item(item)
+    } else {
+      false
+    }) && self.node.cacheable()
```

Comment:
This is a little hard to read. I recommend factoring out a variable for the first boolean.